src/crimson/os/seastore/omap_manager/btree/omap_btree_node_impl.cc

```diff
@@ -195,76 +195,88 @@ OMapInnerNode::rm_key(omap_context_t oc, const std::string &key)
 OMapInnerNode::list_ret
 OMapInnerNode::list(
   omap_context_t oc,
-  const std::optional<std::string> &start,
+  const std::optional<std::string> &first,
+  const std::optional<std::string> &last,
   omap_list_config_t config)
 {
   LOG_PREFIX(OMapInnerNode::list);
-  if (start) {
-    DEBUGT("start={}, this: {}", oc.t, *start, *this);
+  if (first && last) {
+    DEBUGT("first: {}, last: {}, this: {}", oc.t, first, last, *this);
+    assert(first <= last);
+  } else if (first) {
+    DEBUGT("first: {}, this: {}", oc.t, first, *this);
+  } else if (last) {
+    DEBUGT("last: {}, this: {}", oc.t, last, *this);
   } else {
     DEBUGT("this: {}", oc.t, *this);
   }
 
-  auto child_iter = start ?
-    get_containing_child(*start) :
+  auto first_iter = first ?
+    get_containing_child(*first) :
     iter_cbegin();
-  assert(child_iter != iter_cend());
+  auto last_iter = last ?
+    get_containing_child(*last) + 1:
+    iter_cend();
+  assert(first_iter != iter_cend());
 
   return seastar::do_with(
-    child_iter,
-    iter_cend(),
+    first_iter,
+    last_iter,
+    iter_t(first_iter),
     list_bare_ret(false, {}),
-    true,
-    start,
+    first,
+    last,
```

Comment:
first and last don't need to be wrapped by do_with if the responsibility is at the user side.